third_party/move/tools/move-disassembler/src/disassembler.rs

```diff
@@ -1045,11 +1044,9 @@ impl<'a> Disassembler<'a> {
         let locals_names_tys = function_source_map
             .locals
             .iter()
-            .skip(parameter_len)
             .enumerate()
             .map(|(local_idx, (name, _))| {
-                let ty =
-                    self.type_for_local(parameter_len + local_idx, signature, function_source_map)?;
+                let ty = self.type_for_local(local_idx, signature, function_source_map)?;
```

Comment:
It may be better to call `self.type_for_local_or_parameter(parameter_len + local_idx, signature, function_source_map)?;`
so that the indexes 0 as in `L0: ` correspond to 0 in  `MoveLoc[0](...)` in the code listing.

Reply:
If there is at least one param, then `MoveLoc[0]` refers to the first param, not `L0`.

Reply:
There is no `L0`.  It's just a label for a list of locals.  It would be convenient if those labels `L0` or `L2` corresponded to the numbers `[0]` and `[2]` in the code, rather than `L0` corresponding to `[2]` if there are 2 parameters.  `L0` is arbitrary anyway, let's make it arbitrary and useful, rather than arbitrary and confusing.

Reply:
Addressed, as described above, but not using the code suggestion here (because this is not where labels `L0` etc are computed).